src/emit/ethabiencoder.rs

```diff
@@ -654,40 +658,46 @@ impl<'a, 'b> EncoderBuilder<'a, 'b> {
         // We use a little trick here. The length might or might not include the selector.
         // The length will be a multiple of 32 plus the selector (4). So by dividing by 8,
         // we lose the selector.
-        binary.builder.build_call(
-            binary.module.get_function("__bzero8").unwrap(),
-            &[
-                output.into(),
-                binary
-                    .builder
-                    .build_int_unsigned_div(
-                        self.length,
-                        binary.context.i32_type().const_int(8, false),
-                        "",
-                    )
-                    .into(),
-            ],
-            "",
-        );
+        if !self.args.is_empty() {
+            binary.builder.build_call(
+                binary.module.get_function("__bzero8").unwrap(),
+                &[
+                    output.into(),
+                    binary
+                        .builder
+                        .build_int_unsigned_div(
+                            binary.builder.build_int_sub(
+                                self.length,
+                                self.packed_length,
+                                "dyn_len",
+                            ),
+                            binary.context.i32_type().const_int(8, false),
+                            "",
+                        )
+                        .into(),
+                ],
+                "",
+            );
 
-        let mut output = output;
-        let mut offset = self.offset;
-        let mut dynamic = unsafe { binary.builder.build_gep(output, &[self.offset], "") };
+            let mut output = output;
```

Comment:
You can declare the function argument as mutable to avoid creating another variable with the same name.

Reply:
@seanyoung You declared the argument as mutable, but you are still creating the variable here!